tests/shell/environment/env.rs

```diff
@@ -229,3 +229,23 @@ fn std_log_env_vars_have_defaults() {
     assert!(actual.err.contains("%MSG%"));
     assert!(actual.err.contains("%Y-"));
 }
+
+#[test]
+fn env_shlvl() {
+    let actual = nu!("
+        $env.SHLVL = 5
+        nu -i -c 'print $env.SHLVL'
```

Comment:
Thanks for the tests as well! Is the limitation to interactive use universal for the other shells using this quasi standard?

Reply:
Yes. Bash, zsh and fish all work so.
It has been discussed in the related issue.